### ISSUE TITLE
stability: metrics throwing errors in `rho`

### ISSUE THREAD

**Issue by mberhault** [cockroachdb/cockroach#10754]
about 2.5 days after starting up on sha 172ae88, prometheus has started failing scraping `rho` nodes.

Some errors are `connection refused`, some are parsing failures for `/_status/vars` specifically due to this error being surfaced on the page: `MetricFamily has no metrics: name:"sql_mon_admin_session_max" help:"" type:HISTOGRAM`

Similar errors are logged to stderr:
```
E161117 10:47:25.619001 38145322 server/status.go:553  MetricFamily has no metrics: name:"gossip_connections_incoming" help:"" type:GAUGE
E161117 10:47:25.619799 38145322 server.go:2315  http: multiple response.WriteHeader calls
```

rho is also detecting races on the metrics exported:
```
==================
WARNING: DATA RACE
Read at 0x00c4212900b8 by goroutine 1428:
  github.com/cockroachdb/cockroach/pkg/util/metric.(*PrometheusExporter).ScrapeRegistry.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/metric/prometheus_exporter.go:81 +0x277
  github.com/cockroachdb/cockroach/pkg/util/metric.(*Counter).Inspect()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/metric/metric.go:307 +0x49
  github.com/cockroachdb/cockroach/pkg/util/metric.(*PrometheusExporter).ScrapeRegistry()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/metric/prometheus_exporter.go:83 +0x185
  github.com/cockroachdb/cockroach/pkg/server/status.(*MetricsRecorder).scrapePrometheus()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/status/recorder.go:181 +0xe7
  github.com/cockroachdb/cockroach/pkg/server/status.(*MetricsRecorder).PrintAsText()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/status/recorder.go:189 +0x3c
  github.com/cockroachdb/cockroach/pkg/server.(*statusServer).handleVars()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/status.go:551 +0x10f
  github.com/cockroachdb/cockroach/pkg/server.(*statusServer).(github.com/cockroachdb/cockroach/pkg/server.handleVars)-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:702 +0x5f
  net/http.HandlerFunc.ServeHTTP()
      /usr/local/go/src/net/http/server.go:1726 +0x51
  net/http.(*ServeMux).ServeHTTP()
      /usr/local/go/src/net/http/server.go:2022 +0xa1
  github.com/cockroachdb/cockroach/pkg/server.(*Server).ServeHTTP()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:799 +0x165
  net/http.serverHandler.ServeHTTP()
      /usr/local/go/src/net/http/server.go:2202 +0xbb
  net/http.(*conn).serve()
      /usr/local/go/src/net/http/server.go:1579 +0x5f6

Previous write at 0x00c4212900b8 by goroutine 1487:
  [failed to restore the stack]

Goroutine 1428 (running) created at:
  net/http.(*Server).Serve()
      /usr/local/go/src/net/http/server.go:2293 +0x540
  github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func5()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:484 +0x8e
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x8a

Goroutine 1487 (running) created at:
  net/http.(*Server).Serve()
      /usr/local/go/src/net/http/server.go:2293 +0x540
  github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func5()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:484 +0x8e
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x8a
==================
```

The race is due to the rework around a mutex being held while writing to the http io.writer. These still need to be fixed though.

The other errors need to be investigated. This could be a side effect of running the race enabled build, but should investigate further.

Will silence rho for now while I investigate.

**Comment by dianasaur323** [comment]
@mberhault is this still an issue? It's pretty old. If not, we should close out?

**Comment by mberhault** [comment]
Fixed by #14943
